TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -176,8 +176,11 @@ public static IEnumerable<CompletionItem> CreateCompletionItemsForProcedures(IEn
                 ((object[])completionItem.data)[1] = signatureInformation;
 
                 //Store the link between the hash and the procedure. This will help to determine the procedure parameter completion context later. 
-                functionDeclarationSignatureDictionary.Add(signatureInformation, proc);
-                completionItems.Add(completionItem);
+                if (!functionDeclarationSignatureDictionary.ContainsKey(signatureInformation))
+                {
+                    functionDeclarationSignatureDictionary.Add(signatureInformation, proc);
+                    completionItems.Add(completionItem);
+                }
```

Comment:
This means that we only consider the first declaration, if that is what we expect, then ok.

Reply:
@delevoye 
Can you add a test to show the result with the present PR? The test should show that no exception is fired but that there's still a diagnostic issued by the parser because of duplicate procedure declaration (use the test in issue #1687). You don't have to code anything more but the error should already be handled by the parser.

Reply:
@fm-117 
As @smedilol commented in issue #1687, the behavior of Visual Studio has been mimicked: Not adding the element if it is already present.

Reply:
@rooksdo 
Test has been added. As seen with you, it does not show the duplicate procedure declaration because the diagnostic is not created in this case.